src/components/primitives/FormattedNumber.tsx

```diff
@@ -21,14 +22,19 @@ function CompactNumber({ value, visibleDecimals = 2 }: CompactNumberProps) {
     POSTFIXES.length - 1
   );
   const postfix = POSTFIXES[significantDigitsGroup];
-  const formattedValue = normalizeBN(bnValue, 3 * significantDigitsGroup).toNumber();
+  let formattedValue = normalizeBN(bnValue, 3 * significantDigitsGroup).toNumber();
+  if (roundDown) {
+    formattedValue =
+      Math.trunc(Number(formattedValue) * 10 ** visibleDecimals) / 10 ** visibleDecimals;
+  }
```

Comment:
Can we add a comment to this to explain what is happening?

Reply:
Done